File: macos/Sources/NoctusCore/Models/TabState.swift
import Foundation

public struct TabState {
    public let id: Int
    public let location: PathRef
    public var viewMode: ViewMode
    public var sortField: SortField
    public var sortDirection: SortDirection

    public init(id: Int, location: PathRef, viewMode: ViewMode = .details,
                sortField: SortField = .name, sortDirection: SortDirection = .ascending) {
        self.id = id
        self.location = location
        self.viewMode = viewMode
        self.sortField = sortField
        self.sortDirection = sortDirection
    }
}

File: src/Explorer.Core/Models/TabState.cs
namespace NoctusExplorer.Core.Models;

public sealed record TabState(
    int Id,
    PathRef Location,
    ViewMode ViewMode = ViewMode.Details,
    SortField SortField = SortField.Name,
    SortDirection SortDirection = SortDirection.Ascending,
    double? ScrollPosition = null);
